### ISSUE TITLE
`editor.bracketPairColorization.enabled` causes specific JSON to crash

### ISSUE THREAD

**Issue by kuzdogan** [microsoft/vscode#136974]
Issue Type: <b>Performance Issue</b>

I am trying to open a file in VSCode the window freezes when I try to open it. It is a large [hardhat](https://github.com/nomiclabs/hardhat) JSON file but I was able to open other large JSON files. 

I was able to pinpoint the problem to `editor.bracketPairColorization.enabled` setting. When I turn it on it crashes and when I turn it off it fails. I was able to reproduce the problem on vscode.dev. Initially I oculd open the file but turning the sync setting on caused a crash. Switching the setting off there also resolved the problem.

Sharing the file here: https://drive.google.com/file/d/1mxwtEyBEns0Mo2q2u6EBmXCGkm8Ag3n4/view?usp=sharing

VS Code version: Code 1.62.0 (b3318bc0524af3d74034b8bb8a64df0ccf35549a, 2021-11-03T15:18:44.904Z)
OS version: Linux x64 5.4.0-90-generic
Restricted Mode: No

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz (8 x 1000)|
|GPU Status|2d_canvas: enabled<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>oop_rasterization: disabled_off<br>opengl: enabled_on<br>rasterization: disabled_software<br>skia_renderer: enabled_on<br>video_decode: disabled_software<br>vulkan: disabled_off<br>webgl: enabled<br>webgl2: enabled|
|Load (avg)|13, 8, 5|
|Memory (System)|15.50GB (1.81GB free)|
|Process Argv|. --crash-reporter-id 50341597-9d73-4f53-9b48-5bd9ee0113a7|
|Screen Reader|no|
|VM|0%|
|DESKTOP_SESSION|ubuntu|
|XDG_CURRENT_DESKTOP|Unity|
|XDG_SESSION_DESKTOP|ubuntu|
|XDG_SESSION_TYPE|x11|
</details><details>
<summary>Process Info</summary>

```
CPU %	Mem MB	   PID	Process
    1	   143	 22356	code main
    0	    32	 22359	   zygote
    1	   190	 22387	     gpu-process
    0	    16	 22393	       broker
    0	    32	 22360	   zygote
    0	     0	 22362	     zygote
    2	    79	 23074	       window (Issue Reporter)
    0	    48	 22402	   utility
    0	   302	 22416	   window (● c81978245288699b681a6ac399215176.json - hardhat-test - Visual Studio Code)
    0	   222	 22457	     extensionHost
    0	   127	 22551	       electron_node tsserver.js 
    0	   254	 22552	       electron_node tsserver.js 
    0	    79	 22582	         electron_node typingsInstaller.js typesMap.js 
    0	    79	 22604	       /usr/share/code/code /usr/share/code/resources/app/extensions/json-language-features/server/dist/node/jsonServerMain --node-ipc --clientProcessId=22457
    0	    48	 22621	       electron_node eslintServer.js 
    0	    63	 22510	     watcherServiceChokidar
    0	    63	 23023	     searchService
    3	   143	 22456	   shared-process
    0	    63	 22497	     ptyHost
    0	     0	 22675	       /usr/bin/bash
    0	     0	 23092	     /bin/sh -c /usr/bin/ps -ax -o pid=,ppid=,pcpu=,pmem=,command=
    0	     0	 23093	       /usr/bin/ps -ax -o pid=,ppid=,pcpu=,pmem=,command=
```

</details>
<details>
<summary>Workspace Info</summary>

```
|  Window (● c81978245288699b681a6ac399215176.json - hardhat-test - Visual Studio Code)
|    Folder (hardhat-test): 25 files
|      File types: json(8) js(3) gitignore(1) md(1) sol(1)
|      Conf files: package.json(1);
```

</details>
<details><summary>Extensions (15)</summary>

Extension|Author (truncated)|Version
---|---|---
vscode-tailwindcss|bra|0.7.2
better-toml|bun|0.3.2
vscode-eslint|dba|2.2.2
gitlens|eam|11.6.1
vsc-material-theme|Equ|33.1.2
vsc-material-theme-icons|equ|1.2.2
go|gol|0.29.0
vscode-styled-components|jpo|1.6.6
solidity|Jua|0.0.136
vscode-language-babel|mgm|0.0.35
vscode-docker|ms-|1.17.0
soldoc|Nic|0.0.1
code-settings-sync|Sha|3.4.3
indenticator|Sir|0.7.0
vscodeintellicode|Vis|1.2.14

</details><details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vsreu685:30147344
python383:30185418
vspor879:30202332
vspor708:30202333
vspor363:30204092
pythontb:30283811
pythonptprofiler:30281270
vshan820:30294714
vstes263cf:30335440
vscorecescf:30384386
pythondataviewer:30285071
vscod805:30301674
pythonvspyt200:30340761
binariesv615:30325510
vsccppwtct:30382698
bridge0708:30335490
dockerwalkthru:30377721
bridge0723:30353136
pythonrunftest32:30373476
pythonf5test824:30373475
javagetstartedt:30391933
pythonvspyt187:30373474
vsqsis400:30386382
vsaa593cf:30376535
vssld246cf:30386378

```

</details>

**Comment by gjsjohnmurray** [comment]
Perhaps  #136470 which will be fixed in upcoming 1.62.2

**Comment by kuzdogan** [comment]
Oops didn't see as I only searched "JSON crash". Thanks. Closing.
